microstructure_ve.py

```diff
@@ -304,19 +304,51 @@ def load_matlab_microstructure(matfile, var_name):
     return loadmat(matfile, matlab_compatible=True)[var_name]
 
 
+# def assign_intph(microstructure, num_layers):
+#     """Generate interphase layers around the particles.
+
+#     Particles must be falsey, matrix must be truthy
+#     """
+#     from scipy.ndimage import distance_transform_edt
+
+#     dists = distance_transform_edt(microstructure)
+#     particles: np.ndarray = dists == 0
+#     matrix = dists > num_layers
+#     interphase = ~(matrix | particles)
+#     return 2 * matrix.view("u1") + interphase.view("u1")
+
 def assign_intph(microstructure, num_layers):
     """Generate interphase layers around the particles.
-
     Particles must be falsey, matrix must be truthy
     """
     from scipy.ndimage import distance_transform_edt
 
-    dists = distance_transform_edt(microstructure)
+    # special case, matrix with 1 element
+    if len(np.unique(microstructure))==1:
+        return microstructure
```

Comment:
As we discussed, I think this special case should be handled outside this helper function.